s3-vfs/src/main/java/org/pentaho/amazon/s3/S3Util.java

```diff
@@ -42,6 +45,21 @@
   /** Configuration file name */
   public static final String CONFIG_FILE = "config";
 
+  /** Regex for detecting S3 credentials in URI */
+  private static final String URI_AWS_CREDENTIALS_REGEX = "(s3[an]?:\\/)?\\/(?<fullkeys>(?<keys>.*:.*)@)?s3[an]?\\/?(?<bucket>[^\\/]+)(?<path>.*)";
+
+  /** to be used with getKeysFromURI to get the FULL KEYS GROUP **/
+  private static final String URI_AWS_FULL_KEYS_GROUP = "fullkeys";
+
+  /** to be used with getKeysFromURI to get the KEYS GROUP **/
+  private static final String URI_AWS_KEYS_GROUP = "keys";
+
+  /** to be used with getKeysFromURI to get the BUCKET GROUP **/
+  private static final String URI_AWS_BUCKET_GROUP = "bucket";
```

Comment:
Even with sonar issue makes sense keep it because it belongs to same context of regex.